### PR TITLE
[resolves #2560] Migrate to Galleon packs

### LINKED ISSUE #2560: Migrate to Galleon packs
Galleon is the new provisioning mechanism of WildFly that will replace the current feature packs. Based on this http://lists.jboss.org/pipermail/wildfly-dev/2018-June/006573.html message on wildfly-dev list, we should start migrating from FPs to Galleon.

### PR DESCRIPTION
**Pull request by ppalaga**
As noted in #2560 this addresses just the Galleon pack part, keeping the config part open.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,6 +45,9 @@
     <!-- Properties -->
     <properties>
 
+        <!-- Galleon feature pack -->
+        <product.release.name>WildFly Camel</product.release.name>
```
Comment (jamesnetherton) [inline pom.xml]: Do we need this? Can we not use `${project.name}`? Or at a minimum set `product.release.name` to `project.name`.

File: pom.xml [hunk 2]
```diff
@@ -62,7 +65,6 @@
         <version.camel.qpid.proton>0.27.1</version.camel.qpid.proton>
         <version.camel.rxjava2>2.1.16</version.camel.rxjava2>
 
-        <!-- WildFly versions -->
```
Comment (jamesnetherton) [inline pom.xml]: Let's keep this.

File: pom.xml [hunk 3]
```diff
@@ -336,6 +345,11 @@
                     <artifactId>wildfly-feature-pack-build-maven-plugin</artifactId>
```
Comment (jamesnetherton) [inline pom.xml]: Do we still need the `wildfly-feature-pack-build-maven-plugin` plugin and some of the other WF provisioning plugins?

File: itests/standalone/smoke/pom.xml [hunk 4]
```diff
@@ -39,12 +39,27 @@
     </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-spring-javaconfig</artifactId>
```
Comment (jamesnetherton) [inline itests/standalone/smoke/pom.xml]: Why do we need these new dependencies?

File: feature/extrasA/wildfly-feature-pack-build.xml [hunk 5]
```diff
@@ -0,0 +1,43 @@
+<!--
+  ~ JBoss, Home of Professional Open Source.
```
Comment (jamesnetherton) [inline feature/extrasA/wildfly-feature-pack-build.xml]: Maybe use the WFC project license? Likewise for the other `feature-pack-build.xml `files.

File: pom.xml (hunk 1 above)
Comment (ppalaga) [reply pom.xml]: IIRC, this string appears on the console, when the server starts. It needs to be different and easily maintainable for Fuse@EAP. Would `${project.name}` work well also for the product?

File: pom.xml (hunk 2 above)
Comment (ppalaga) [reply pom.xml]: Unintentional. Sorry.

File: pom.xml (hunk 1 above)
Comment (jamesnetherton) [reply pom.xml]: Ok, fair enough. On the product side we wont want to use `project.name` so I guess we need this separate property.

File: pom.xml (hunk 3 above)
Comment (ppalaga) [reply pom.xml]: No, I thought I removed those. Let me re-check.

File: itests/standalone/smoke/pom.xml (hunk 4 above)
Comment (ppalaga) [reply itests/standalone/smoke/pom.xml]: They were needed two months ago, but now when re-checking they are not needed anymore. Apparently the WF Galleon pack brings them now.

File: feature/extrasA/wildfly-feature-pack-build.xml (hunk 5 above)
Comment (ppalaga) [reply feature/extrasA/wildfly-feature-pack-build.xml]: +1 and I wish we could introduce mycila license plugin which these kind of issues can be checked and fixed automatically.

Comment (jamesnetherton) [reply feature/extrasA/wildfly-feature-pack-build.xml]: We do have the [license-maven-plugin](https://github.com/jamesnetherton/wildfly-liquibase/blob/master/pom.xml#L187).

Comment (ppalaga) [reply feature/extrasA/wildfly-feature-pack-build.xml]: Apparently, it does not protest on missing/wrong license headers. I know and prefer com.mycila:license-maven-plugin.

Comment (jamesnetherton) [reply feature/extrasA/wildfly-feature-pack-build.xml]: I think it does. Via the `check-file-header` goal, I use it in one of my personal projects.

**Comment by ppalaga** [comment]
>  why so much stuff has been commented out

My fault, sorry. All junk should be removed now, except for the license.xml stuff.

**Comment by ppalaga** [comment]
d875d12 should resolve all things found by @jamesnetherton . If this state is acceptable, I'd create an issue for the license.xml stuff to solve it later.

File: feature/extrasA/wildfly-feature-pack-build.xml (hunk 5 above)
Comment (ppalaga) [reply feature/extrasA/wildfly-feature-pack-build.xml]: `check-file-header` does not seem to be active by default.

**Comment by jamesnetherton** [comment]
Mostly looks good. Just a couple of final things...

1. I still see `wildfly-feature-pack-build-maven-plugin` in the root pom.xml `pluginManagement` do we need it?

2. Let's raise the license.xml issue now and reference it in a comment above the profile that you commented out. Just so that it's obvious why it has been deactivated.

**Comment by ppalaga** [comment]
> 1. I still see `wildfly-feature-pack-build-maven-plugin` in the root pom.xml `pluginManagement` do we need it?
>
> 2. Let's raise the license.xml issue now and reference it in a comment above the profile that you commented out. Just so that it's obvious why it has been deactivated. 

Both done in 5a600d9

**Comment by jamesnetherton** [comment]
I think the wrong commit just got merged.

Every time you push changes, it'll queue up PR builds for each commit. If you remove the 'do not merge' label before the last PR build is complete, there's a chance it will auto-merge a stale commit. I think this is what just happened.

This would all be avoidable if we dropped the auto-merge thing and just used the merge button.

I'll back out the incorrect commit and we can retest this PR.

**Comment by jamesnetherton** [comment]
retest this please

**Comment by ppalaga** [comment]
> Every time you push changes, it'll queue up PR builds for each commit. If you remove the 'do not merge' label before the last PR build is complete, there's a chance it will auto-merge a stale commit.

Sorry, I thought, any jobs building from outdated commits are cancelled after pushing to a PR branch. Will keep the info in mind.

> This would all be avoidable if we dropped the auto-merge thing and just used the merge button.

I do not mind manual merges.

**Pull request merged**